router/pkg/authentication/authentication.go

```diff
@@ -53,6 +55,20 @@ func (a *authentication) Claims() Claims {
 	return a.claims
 }
 
+func (a *authentication) SetScopes(scopes []string) {
+	if a == nil {
+		return
+	}
+	if a.claims == nil {
+		a.claims = make(Claims)
+	}
+	existingScopes := a.claims["scope"]
+	if existingScopes != nil && existingScopes.(string) != "" {
```

Comment:
Question for reviewers: should this function overwrite the existing scopes, or set them in addition? I had assumed that it would set them in addition

Reply:
Good question. I'd leave it up to the user, if he wants to copy the current values before setting the new values. So it should not append.